src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -163,25 +161,16 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
             CancellationToken cancellationToken)
         {
             var tokens = await FindMatchingIdentifierTokensAsync(state, identifier, cancellationToken).ConfigureAwait(false);
-
-            return await FindReferencesInTokensAsync(
-                symbol,
-                state,
-                tokens,
-                static (state, token, identifier, _) => IdentifiersMatch(state.SyntaxFacts, identifier, token),
```

Comment:
this was hte general issue i wanted to avoid. We already call FindMatchingIdentifierTokensAsync above.  so it's redundant to check if the identifiers match again here.  So we just have to pass along these pointless predicates that either do redundant work, do nothing, or assert that we're getting the right tokens.  Just cleaner to avoid all of that.